test/mlang/strformat.mc

```diff
@@ -0,0 +1,81 @@
+// String format with MLang
+
+include "char.mc"
+include "seq.mc"
+include "string.mc"
+
+lang TypeInteger
+	syn Type =
+	| TyInt (Int)
+
+	sem toString =
+	| TyInt v -> int2string v
+end
```

Comment:
I'm wondering if you should choose names that is more specific for this particular library. The name `Type` is very general, and without looking at the entire library it isn't obvious why something denoting a type should also carry with it a value of that type. How about changing `Type` to `Fmt`:

```
lang FmtInteger
  syn Fmt =
  | FmtInt Int

  sem toString =
  | FmtInt n -> int2string n
end
```

Reply:
The though process here was more that all types probably should have some form of toString and that they should be easy to combine with the StrFormatBase language. Changed it to Fmt now.